### PR TITLE
[BugFix]Change the RapidsDiskBlockManager in ShuffleBufferCatalog to guarantee the shuffle files can be cleaned successfully

### PR DESCRIPTION
**Pull request by zengruios**
Fix bug described in issue https://github.com/NVIDIA/spark-rapids/issues/6240.

Signed-off-by: zengruios <578395184@qq.com>

### CODE REVIEW

**Comment by zengruios** [comment]
@revans2 @abellina help to review the new PR.

**Comment by abellina** [comment]
Please note that just a single commit needs to be signed, so the other PR could have been left open (https://github.com/NVIDIA/spark-rapids/pull/6242) and a second signed commit could have been pushed. Not a big deal, and thanks for the contribution.

**Comment by abellina** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferCatalog.scala [hunk 1]
```diff
@@ -298,4 +299,6 @@ object RapidsBufferCatalog extends Logging with Arm {
 
   /** Remove a buffer ID from the catalog and release the resources of the registered buffer. */
   def removeBuffer(id: RapidsBufferId): Unit = singleton.removeBuffer(id)
+
+  def getDiskBlockManager(): RapidsDiskBlockManager = diskBlockManager
```
Comment (abellina) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferCatalog.scala:303]: I had commented on this nit in the previous PR, but it is minor and not a blocker.
```suggestion
  def getDiskBlockManager: RapidsDiskBlockManager = diskBlockManager
```

**Review by abellina — APPROVED** [review]

**Review by revans2 — APPROVED** [review]

**Comment by abellina** [comment]
Thanks @zengruios

**Pull request merged**